pytradfri/api/aiocoap_api.py

```diff
@@ -43,15 +43,43 @@ def api_factory(host, security_code, loop=None):
     if loop is None:
         loop = asyncio.get_event_loop()
 
-    security_code = security_code.encode('utf-8')
+    PatchedDTLSSecurityStore.SECRET_PSK = security_code.encode('utf-8')
 
-    PatchedDTLSSecurityStore.SECRET_PSK = security_code
+    _protocol = None
 
     @asyncio.coroutine
     def _get_protocol():
         """Get the protocol for the request."""
-        protocol = yield from Context.create_client_context(loop=loop)
-        return protocol
+        nonlocal _protocol
+        if not _protocol:
+            _protocol = yield from Context.create_client_context(loop=loop)
```

Comment:
Since we know that we will always do a test request, why not create it right away? Every yield, even if it is just to fetch a variable, will suspend the task and wait for the other task to return the result.